edge/client/portainer_edge_async_client.go

```diff
@@ -415,14 +415,7 @@ func (client *PortainerAsyncClient) executeAsyncRequest(payload AsyncRequest, po
 	defer resp.Body.Close()
 
 	if resp.StatusCode != http.StatusOK {
-		errorData := parseError(resp)
-		logError(resp, errorData)
-
-		if errorData != nil {
-			return nil, errors.New(errorData.Message + ": " + errorData.Details)
-		}
-
-		return nil, errors.New("short poll request failed")
+		return nil, logPollingError(resp, "AsyncEdgeAgentExecuteAsyncRequest", fmt.Sprintf("AsyncEdgeAgent [%d] failed to execute async request", client.getEndpointIDFn()))
```

Comment:
Please try to avoid using `Sprintf()` for this purpose, it's much better to use the structured logging functions that zlog provides, in this case something like `.Int("endpoint_id", endpointID)`.